cognite/client/data_classes/data_modeling/data_models.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import annotations
+
+import json
+from typing import TYPE_CHECKING, Any, List, Literal, Union, cast
+
+from cognite.client.data_classes._base import (
+    CogniteFilter,
+    CogniteResourceList,
+)
+from cognite.client.data_classes.data_modeling._validation import validate_data_modeling_identifier
+from cognite.client.data_classes.data_modeling.shared import DataModeling, ViewReference
+from cognite.client.data_classes.data_modeling.views import View, ViewApply
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+
+
+class DataModelCore(DataModeling):
+    """A group of views.
+
+    Args:
+        space (str): The workspace for the data model, a unique identifier for the space.
+        external_id (str): Combined with the space is the unique identifier of the data model.
+        description (str): Textual description of the data model
+        name (str): Human readable name for the data model.
+        version (str): DMS version.
+    """
+
+    def __init__(
+        self,
+        space: str = None,
+        external_id: str = None,
```

Comment:
These aren't optional.
```suggestion
        space: str,
        external_id: str,
```

Reply:
Data will break the _load method for CogniteResource._load method:
```
    @classmethod
    def _load(
        cls: Type[T_CogniteResource], resource: Union[Dict, str], cognite_client: CogniteClient = None
    ) -> T_CogniteResource:
        if isinstance(resource, str):
            return cls._load(json.loads(resource), cognite_client=cognite_client)
        elif isinstance(resource, Dict):
            instance = cls(cognite_client=cognite_client)
            for key, value in resource.items():
                snake_case_key = to_snake_case(key)
                if hasattr(instance, snake_case_key):
                    setattr(instance, snake_case_key, value)
            return instance
        raise TypeError(f"Resource must be json str or dict, not {type(resource)}")
```